File: oop/ad.cpp
#include "base_model.cpp"

class Ad : public BaseAdvertising {
  private:
    string title;
    string imgUrl;
    string link;
    Advertiser* owner = NULL;
  public:
    Ad(int _id, string _imgUrl, string _link, Advertiser* _owner) : BaseAdvertising(_id) {
      this->title  = "no-title" ;
      this->imgUrl = _imgUrl;
      this->link   = _link  ;
      this->owner  = _owner ;
    }

    string getLink(){
      return this->link;
    }
    void setLink(string x){
      this->link = x;
    }

    string getImgUrl(){
      return this->imgUrl;
    }
    void setImgUrl(string x){
      this->imgUrl = x;
    }

    void setAdvertiser(Advertiser* x) {
      this->owner = x;
    }

    void describeMe() {
      cout << "I am an Ad\n";
    }

    void incClicks() {
      this->BaseAdvertising::incClicks();
      this->owner->incClicks();
    }
    void incViews() {
      this->BaseAdvertising::incViews();
      this->owner->incViews();
    }
};

File: python/ad.py
from base_model import *

class Ad(BaseAdvertising):
  
  def __init__(self, id, imgUrl, link, owner):
    super().__init__(id)
    self.__title  = "no-title"
    self.__imgUrl = imgUrl
    self.__link   = link
    self.__owner  = owner

  def getTitle(self):
    return self.__title
  def setTitle(self, x):
    self.__title = x

  def getImgUrl(self):
    return self.__imgUrl
  def setImgUrl(self, x):
    self.__imgUrl = x

  def getLink(self):
    return self.__link
  def setLink(self, x):
    self.__link = x

  def setAdvertiser(self, x):
    self.__owner = x

  def incClicks(self):
    super().incClicks()
    self.__owner.incClicks()

  def incViews(self):
    super().incViews()
    self.__owner.incViews()